Testing/functional/tests/cypress/support/commands.js

```diff
@@ -128,85 +132,116 @@ Cypress.Commands.add(
-                                    globalStorage.authCookies = cookies;
-                                }
-                            );
-                        });
+                            .then((response) => {
+                                cy.log("Posting credentials");
+                                const html = document.createElement("html");
+                                html.innerHTML = response.body;
+                                const form =
+                                    html.getElementsByTagName("form")[0];
+                                const url = form.action;
+                                return cy.request({
+                                    method: "POST",
+                                    url,
+                                    followRedirect: false,
+                                    form: true,
+                                    body: {
+                                        username: username,
+                                        password: password,
+                                    },
+                                });
+                            })
+                            .then((response) => {
+                                let callBackQS = response.headers["location"];
+                                const callbackURL = `${callBackQS}`;
+                                cy.log(
+                                    `Visiting Callback ${callBackQS}`,
+                                    response
+                                );
+                                cy.visit(callbackURL, { timeout: 60000 });
+                                // store auth cookies
+                                cy.getCookies({ timeout: 60000 }).then(
+                                    (cookies) => {
+                                        globalStorage.authCookies = cookies;
+                                    }
+                                );
+                            });
+                    });
                 });
+
+                // Setup standard aliases for busy endpoint calls
+                setupStandardAliases();
+
+                cy.log(`Visit path: ${path}`);
+                cy.visit(path, { timeout: 60000 });
+
+                if (configSettings === null) {
+                    cy.readConfig().then((config) => {
+                        cy.log(`Read config: ${JSON.stringify(config)}`);
+                        // Make sure to wait on busy endpoint calls
+                        waitForInitialDataLoad(username, config, path);
+                    });
+                } else {
+                    cy.log(`Session config: ${JSON.stringify(configSettings)}`);
+                    // Make sure to wait on busy endpoint calls
+                    waitForInitialDataLoad(
+                        username,
+                        JSON.parse(configSettings),
+                        path
+                    );
+                }
```

Comment:
When configSettings is null, it may be beneficial to call cy.readConfig() before cy.visit(), so I'd suggest moving the cy.visit() from outside the if statement into each branch.

Reply:
Changed.